src/connect.spec.ts

```diff
@@ -296,4 +297,22 @@ describe('devtools connect', () => {
       await client.close();
     });
   });
+
+  describe('oidcHasProviderFlow', function() {
+    it('returns false by default', function() {
+      expect(oidcHasProviderFlow('mongodb://example/', {})).to.equal(false);
+    });
+
+    it('returns true if the PROVIDER_NAME JS option is set', function() {
+      expect(oidcHasProviderFlow('mongodb://example/', {
+        authMechanismProperties: {
+          PROVIDER_NAME: 'aws'
+        }
+      })).to.equal(true);
+    });
+
+    it('returns true if the PROVIDER_NAME url option is set', function() {
+      expect(oidcHasProviderFlow('mongodb://example/?authMechanismProperties=PROVIDER_NAME:aws', {})).to.equal(true);
```

Comment:
Should this only happen when authMechanism=`MONGODB-OIDC`? That might be an edge case we probably don't need to worry.

Reply:
Hm yeah. It shouldn't make a difference right now anyway, because for other auth mechanisms, the callbacks are being ignored anyway